tests/executables_src/testLMapMathPluginPushPars.cc

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE LMapMathPluginTest
+#include <boost/test/unit_test.hpp>
+using namespace boost::unit_test_framework;
+
+#include "Device.h"
+#include "LogicalNameMappingBackend.h"
+
+using namespace ChimeraTK;
+
+BOOST_AUTO_TEST_SUITE(LMapMathPluginTestSuite)
+
+/********************************************************************************************************************/
+
+struct DummyForCleanupCheck : public LogicalNameMappingBackend {
+  using LogicalNameMappingBackend::LogicalNameMappingBackend;
+
+  static boost::shared_ptr<DeviceBackend> createInstance(std::string, std::map<std::string, std::string>) {
+    return boost::make_shared<DummyForCleanupCheck>("mathPluginWithPushPars.xlmap");
+  }
+  ~DummyForCleanupCheck() override {
+    std::cout << "~DummyForCleanupCheck()" << std::endl;
+    cleanupCalled = true;
+  }
+
+  struct BackendRegisterer {
+    BackendRegisterer() {
+      ChimeraTK::BackendFactory::getInstance().registerBackendType(
+          "DummyForCleanupCheck", &DummyForCleanupCheck::createInstance, {"map"});
+    }
+  };
+  static std::atomic_bool cleanupCalled;
+};
+std::atomic_bool DummyForCleanupCheck::cleanupCalled;
+static DummyForCleanupCheck::BackendRegisterer gDFCCRegisterer;
+
+BOOST_AUTO_TEST_CASE(testPushPars) {
+  setDMapFilePath("mathPluginWithPushPars.dmap");
+  {
+    // we enforce that initial values from variable definition in xlmap are never used in MathPugin
+    // - therefore, in test here, we must issue write for all push-parameters
+    // - then, version number check in MathPlugin will notice valid data has been provided after open
+    // - for poll-type accessors, version number check also succeeds since they get recent version numbers anyway
+    // Test needs to call activateAsyncRead but it must be irrelevant whether before or after writes
+
+    ChimeraTK::Device targetDevice;
+    targetDevice.open("HOLD");
+
+    auto accTarget = targetDevice.getScalarRegisterAccessor<uint32_t>("HOLD0/WORD_G");
+    auto pollPar = targetDevice.getScalarRegisterAccessor<uint32_t>("HOLD0/POLLPAR");
+    pollPar = 1;
+    pollPar.write();
+
+    ChimeraTK::Device device;
+    device.open("EOD");
+    device.activateAsyncRead();
+    auto pushPar = device.getScalarRegisterAccessor<uint32_t>("DET/EXPOSURE");
+
+    pushPar = 2;
+    pushPar.write();
+
+    auto accMathWrite = device.getScalarRegisterAccessor<double>("DET/GAIN");
+    // we don't have main value (x in formula) yet, since it wasn't yet written.
+    // therefore, we expect to have no value yet for formula output (0 is default from dummy construction)
+    accTarget.readLatest();
+    BOOST_CHECK_EQUAL(int(accTarget), 0);
+
+    // write to main value and check result
+    accMathWrite = 3;
+    accMathWrite.write();
+    accTarget.readLatest();
+    BOOST_CHECK_EQUAL(int(accTarget), 100 * (int)pollPar + 10 * (int)pushPar + (int)accMathWrite);
```

Comment:
don't use C-type casts!

Reply:
Why do you cast at all? The implicit conversion  should do the job for the multiplication and addition. int(accTarget) is the only needed one and proper C++ already.

Reply:
You have to cast accessors in BOOST_CHECK_EQUAL etc, otherwise you get some errors because BOOST tries to cast them into a string for the message, which does not work. It is likely sufficient though to do that for the accessor which is not part of any expression (`accTarget` in this case).

I also recommend to use `BOOST_TEST` instead (although it has the same issue with casting):

```
BOOST_TEST(int(accTarget) ==  100 * pollPar + 10 * pushPar + accMathWrite);
```

Reply:
changed it to use BOOST_TEST